cmd/cluster/destroy.go

```diff
@@ -104,53 +109,89 @@ func NewDestroyCommand() *cobra.Command {
 func DestroyCluster(ctx context.Context, o *DestroyOptions) error {
 	c := util.GetClientOrDie()
 
+	infraID := o.InfraID
+	baseDomain := o.BaseDomain
+	region := o.Region
+
+	hostedClusterExists := false
 	var hostedCluster hyperv1.HostedCluster
 	if err := c.Get(ctx, types.NamespacedName{Namespace: o.Namespace, Name: o.Name}, &hostedCluster); err != nil {
-		log.Info("hostedcluster not found, nothing to do", "namespace", o.Namespace, "name", o.Name)
-		return nil
+		log.Info("Hosted cluster not found", "namespace", o.Namespace, "name", o.Name)
+	} else {
+		infraID = hostedCluster.Spec.InfraID
+		baseDomain = hostedCluster.Spec.DNS.BaseDomain
+		region = hostedCluster.Spec.Platform.AWS.Region
+		hostedClusterExists = true
+		log.Info("Found hosted cluster", "namespace", hostedCluster.Namespace, "name", hostedCluster.Name, "infraID", infraID, "region", region, "baseDomain", baseDomain)
 	}
 
-	controllerutil.AddFinalizer(&hostedCluster, destroyFinalizer)
-	if err := c.Update(ctx, &hostedCluster); err != nil {
-		return fmt.Errorf("failed to add finalizer, won't destroy: %w", err)
+	// If the hosted cluster isn't around, fall back to user inputs
+
+	if !hostedClusterExists {
```

Comment:
These are errors regardless of whether or not `hostedClusterExists` right?

Reply:
I guess I could argue that for an AWS hostedcluster these fields are required and so the object would be invalid if any of them were empty, but I guess it wouldn't hurt to just unconditionally validate. However, maybe worth just changing in a followup the next time we're here rather than induce another e2e run? Let me know what you think

Reply:
Tests bombed out so I went ahead and fixed this here